src/operation.rs

```diff
@@ -348,6 +365,68 @@ impl Operation {
         //TODO: IF ERROR, RETURN AN Operation THAT CAN UNDO THE CURRENT STATE
         //TODO: SAFELY HANDLE CANCEL
         match self {
+            Self::Compress {
+                paths,
+                to,
+                archive_type,
+            } => {
+                tokio::task::spawn_blocking(move || -> Result<(), String> {
+                    match archive_type {
+                        ArchiveType::Zip => {
+                            let mut archive = fs::File::create(&to)
+                                .map(io::BufWriter::new)
+                                .map(zip::ZipWriter::new)
+                                .map_err(err_str)?;
+
+                            let zip_options = zip::write::SimpleFileOptions::default();
+
+                            let mut paths = paths;
+                            for path in paths.clone().iter() {
+                                if path.is_dir() {
+                                    let new_paths_it = WalkDir::new(path).into_iter();
+                                    for entry in new_paths_it.skip(1) {
+                                        let entry = entry.map_err(err_str)?;
+                                        paths.push(entry.path().to_path_buf());
+                                    }
+                                }
+                            }
+
+                            for path in paths {
```

Comment:
Can progress be reported? Currently it just goes from 0% straight to 100% when it finishes.